File: server/Models/Entities/Submission.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models.Entities
{
    public class Submission
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        public Guid ThesisId { get; set; }
   
        [Required]
        public string? StudentNumber { get; set; }

        [Required]
        public DateTime SubmittedAt { get; set; }    

        [MaxLength(999)]
        public string? Notes { get; set; }
        
        // Navigation Properties
        public Thesis? Thesis { get; set; }
        public Student? Student { get; set; }
    }
}

File: packages/types/src/submission.ts
export interface CreateSubmissionDto {
  thesisId: string;
  studentNumber: string;
  notes?: string;
}

export interface UpdateSubmissionDto {
  notes?: string;
}

export interface SubmissionResponseDto {
  id: string;
  thesisId: string;
  studentNumber: string;
  submittedAt: string;
  notes?: string;
  thesisTitle?: string;
  studentName?: string;
}

export type SubmissionResponseListDto = SubmissionResponseDto[];